models/productDao.js

```diff
@@ -0,0 +1,61 @@
+const { teaDataSource } = require("./datasource.js");
+
+const getProduct = async (start, pageSize, category, sub, type, sort) => {
+  let whereCondition = category ? `WHERE` : type ? `WHERE` : "";
+  let andCondition = category && type ? `AND` : sub && type ? `AND` : "";
+  const categoryCondition = category ? `c.name='${category}'` : "";
+  const subCondition = sub ? `AND s.name='${sub}'` : "";
+  const typeCondition = type[0]
+    ? type[1]
+      ? type[2]
+        ? type[3]
+          ? `(t.type='${type[0]}' OR t.type='${type[1]}' OR t.type='${type[2]}',t.type='${type[3]}')`
+          : `(t.type='${type[0]}' OR t.type='${type[1]}' OR t.type='${type[2]}')`
+        : `(t.type='${type[0]}' OR t.type='${type[1]}')`
+      : `(t.type='${type[0]}')`
+    : "";
```

Comment:
우선 삼항연산자를 쓰는 이유는 다음과 같습니다:
- 조건문의 조건이 짧고 `if ... else ...`로 조건이 분기처리되는 경우 if else 문을 사용하지 않고 더 간결하게 작성하기 위해

하지만, 지금 삼항연산자를 통해 더욱 더 가독성이 떨어지고 있는 것 같습니다.

그렇기 때문에 저번주에 간략하게 언급드린 `Query builder`를 사용하게 됩니다.
쿼리 빌더는 함수로 혹은 객체 매핑을 통해 구현할 수 있습니다. (세부 사항은 13일 월요일 쿼리 빌더 세션 간 전달 예정)
쿼리 빌더는 Data Access Layere에서 사용되는 raw query문을 반환하기 때문에 서비스 로직 혹은 각 dao 파일에서 작성해도 되고, 쿼리 빌더가 길어지는 경우 하나의 모듈로 분리할 수도 있으니 참고해주세요! :)


예시는 다음과 같습니다:

```suggestion
  let conditionClause = "";
  
  if ( categoryCondition) {
    conditionCluase += `WHERE c.name = ${category}`;
  }
```